hazelcast/serialization/output.py

```diff
@@ -30,31 +50,55 @@ def write_from(self, buff, offset=None, length=None):
         elif length == 0:
             return
         self._ensure_available(_len)
-        self._buffer[self._pos : self._pos + _len] = buff
+        self._buffer[self._pos : self._pos + _len] = buff[_off : _off + _len]
         self._pos += _len
 
     def write_boolean(self, boolean):
         self._write(1 if boolean else 0)
 
+    def write_boolean_bit_positional(self, val: bool, pos: int, bit_pos: int) -> None:
+        b = self._buffer[pos]
+        if val:
+            b = b | (1 << bit_pos)
+        else:
+            b = b & ~(1 << bit_pos)
+        self._buffer[pos] = b
+
     def write_byte(self, val):
         self._write(val)
 
+    def write_signed_byte(self, val: int) -> None:
+        self._ensure_available(BYTE_SIZE_IN_BYTES)
+        self._FMT_SIGNED_BYTE.pack_into(self._buffer, self._pos, val)
+        self._pos += BYTE_SIZE_IN_BYTES
+
+    def write_signed_byte_positional(self, val: int, pos: int) -> None:
```

Comment:
Should we throw if position is out of buffer? Currently, this code will do nothing silently.

Reply:
No, the `pack_into` call would throw for out of array positions

Reply:
Sorry I was not changing the init_size but removing write_zero_bytes and running the test. You're right.